File: common/src/main/java/juuxel/adorn/item/ItemWithDescription.java
package juuxel.adorn.item;

import net.minecraft.item.Item;
import net.minecraft.item.ItemStack;
import net.minecraft.item.tooltip.TooltipType;
import net.minecraft.text.Text;
import net.minecraft.util.Formatting;

import java.util.List;

public class ItemWithDescription extends Item {
    public ItemWithDescription(Settings settings) {
        super(settings);
    }

    protected String getDescriptionKey(ItemStack stack) {
        return getTranslationKey(stack) + ".description";
    }

    @Override
    public void appendTooltip(ItemStack stack, TooltipContext context, List<Text> tooltip, TooltipType type) {
        super.appendTooltip(stack, context, tooltip, type);
        tooltip.add(createDescriptionText(getDescriptionKey(stack)));
    }

    public static Text createDescriptionText(String translationKey) {
        return Text.translatable(translationKey)
            .styled(style -> style.withItalic(true).withColor(Formatting.DARK_GRAY));
    }
}

File: common/src/main/kotlin/juuxel/adorn/item/ItemWithDescription.kt
package juuxel.adorn.item

import net.minecraft.client.item.TooltipContext
import net.minecraft.item.ItemStack
import net.minecraft.text.Text
import net.minecraft.text.TranslatableText
import net.minecraft.util.Formatting
import net.minecraft.world.World

open class ItemWithDescription(settings: Settings) : SimpleAdornItem(settings) {
    override fun appendTooltip(stack: ItemStack, world: World?, texts: MutableList<Text>, context: TooltipContext) {
        super.appendTooltip(stack, world, texts, context)
        texts.add(createDescriptionText("$translationKey.desc"))
    }

    companion object {
        fun createDescriptionText(translationKey: String): Text =
            TranslatableText(translationKey).styled {
                it.withItalic(true).withColor(Formatting.DARK_GRAY)
            }
    }
}
